libsolidity/ast/Types.h

```diff
@@ -208,14 +208,30 @@ class Type
 	virtual bool operator==(Type const& _other) const { return category() == _other.category(); }
 	virtual bool operator!=(Type const& _other) const { return !this->operator ==(_other); }
 
-	/// @returns number of bytes used by this type when encoded for CALL. If it is a dynamic type,
-	/// returns the size of the pointer (usually 32). Returns 0 if the type cannot be encoded
-	/// in calldata.
+	/// @returns number of bytes used by this type when encoded for CALL. Cannot be used for
+	/// dynamically encoded types.
+	/// Always returns a value greater than zero and throws, if the type cannot be encoded in calldata
+	/// (or is dynamically encoded).
 	/// If @a _padded then it is assumed that each element is padded to a multiple of 32 bytes.
-	virtual unsigned calldataEncodedSize(bool _padded) const { (void)_padded; return 0; }
+	virtual unsigned calldataEncodedSize(bool _padded) const { (void)_padded; solAssert(false, ""); }
+	/// @returns the distance between two elements of this type in an array, tuple or struct.
+	/// For statically-sized types this is the same as calldataEncodedSize(true).
```

Comment:
statically-size or statically encoded? The term `statically-sized` only exists for arrays, doesn't it?

Reply:
Ok - I'll change it to ``statically encoded`` (and remove the hyphen in ``dynamically-encoded`` below in the process).